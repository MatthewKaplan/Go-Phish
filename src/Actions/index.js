export const allMembers = members => ({
  type: "ALL_MEMBERS",
  members
});

export const currentMember = member => ({
  type: "CURRENT_MEMBER",
  member
});

export const allYears = years => ({
  type: "ALL_YEARS",
  years
});

export const allTours = tours => ({
  type: "ALL_TOURS",
  tours
});

export const allSongs = songs => ({
  type: "ALL_SONGS",
  songs
});

export const allVenues = venues => ({
  type: "ALL_VENUES",
  venues
});

export const currentShows = shows => ({
  type: "CURRENT_SHOWS",
  shows
});
