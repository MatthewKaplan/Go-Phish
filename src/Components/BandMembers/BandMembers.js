import React, { Component } from "react";
import './BandMembers.scss';

class BandMembers extends Component {
  render() {
    const { member } = this.props;

    console.log(member);

    const memberImg = {
      backgroundImage: `url(${member.image})`
    };

    return (
      <div className="band-members-component">
        <section className="member" style={memberImg} />
          <h1>{member.member}</h1>
      </div>
    );
  }
}

export default BandMembers;
