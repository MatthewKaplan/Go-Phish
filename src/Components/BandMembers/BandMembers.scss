.band-members-component {
  display: inline-block;
  margin: 4px;
  position: relative;
  
  .member {
    border-radius: 4px;
    background-repeat: no-repeat;
    background-size: cover;
    height: 550px;
    width: 650px;
    margin: 50px auto;
    cursor: pointer;
    transition: .3s;
  }
}

.band-members-component :not(section) {
  position: absolute;
  top: 50%;
  z-index: 1;
  color: #fff;
  text-align: center;
  width: 100%;
  opacity: 0;
  transition: .3s;
  font-size: 55px;
}

.band-members-component:hover :not(section) {
  opacity: 1;
}

.band-members-component:hover section {
  filter:blur(2px) brightness(60%);
  box-shadow: 0 0 16px #772954;
}