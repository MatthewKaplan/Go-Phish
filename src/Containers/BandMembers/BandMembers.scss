.band-member-cards {
  display: inline-block;
  margin: 4px;
  position: relative;
  
  .member {
    border-radius: 4px;
    background-repeat: no-repeat;
    background-size: cover;
    height: 550px;
    width: 650px;
    margin: 50px auto;
    cursor: pointer;
    transition: .3s;
    margin-bottom: 25px;
  }
}

.band-member-cards :not(section) {
  position: absolute;
  top: 50%;
  z-index: 1;
  color: #fff;
  text-align: center;
  width: 100%;
  opacity: 0;
  transition: .3s;
  font-size: 55px;
}

.band-member-cards:hover :not(section) {
  opacity: 1;
  transition: 1s;
}

.band-member-cards:hover section {
  filter:blur(2px) brightness(60%);
  box-shadow: 0 0 16px #772954;
}

.member-info {
  position: relative;
}