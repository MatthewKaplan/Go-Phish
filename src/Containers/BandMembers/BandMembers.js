import React, { Component } from "react";
import { connect } from "react-redux";
import { currentMember } from "../../Actions/index";
import PropTypes from "prop-types";

export class BandMembers extends Component {
  getCurrentMember = member => {
    this.props.currentMember(member);
  };

  render() {
    const { member } = this.props;
    const memberImg = {
      backgroundImage: `url(${member.image})`
    };

    return (
      <div className="band-member-component">
        <div
          className="band-member-cards"
          data-test="band-member-cards"
          onClick={() => this.getCurrentMember(member)}
        >
          <section className="member" style={memberImg} />
          <h1>{member.member}</h1>
        </div>
      </div>
    );
  }
}

BandMembers.propTypes = {
  member: PropTypes.array
};

export const mapDispatchToProps = dispatch => ({
  currentMember: member => dispatch(currentMember(member))
});

export default connect(
  null,
  mapDispatchToProps
)(BandMembers);
