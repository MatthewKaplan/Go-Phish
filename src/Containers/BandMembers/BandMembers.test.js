import React from "react";
import { shallow } from "enzyme";
import MockData from "../../Helpers/mockData";
import * as actions from "../../Actions/index";
import { BandMembers, mapDispatchToProps } from "./BandMembers";

const member = MockData.mockMember;
let mockCurrentMember = jest.fn();

describe("BandMembers", () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(
      <BandMembers member={member} currentMember={mockCurrentMember} />
    );
    instance = wrapper.instance();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke 'getCurrentMember' with correct params when a band member card is clicked", () => {
    jest.spyOn(instance, "getCurrentMember");
    wrapper.find("[data-test='band-member-cards']").simulate("click");
    expect(instance.getCurrentMember).toHaveBeenCalledWith(member);
  });
});

describe("mapDispatchToProps", () => {
  it("should call dispatch for currentMember", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = actions.currentMember(member);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.currentMember(member);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
