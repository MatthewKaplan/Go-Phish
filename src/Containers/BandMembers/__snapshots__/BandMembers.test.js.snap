// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BandMembers should match the snapshot 1`] = `ShallowWrapper {}`;
